orderbook/src/database.rs

```diff
@@ -62,6 +66,19 @@ impl Postgres {
     }
 }
 
+pub trait DataRetrieving: PreSignatureRetrieving + TradeRetrieving {
+    fn as_presignature_retrieving(self: Arc<Self>) -> Arc<dyn PreSignatureRetrieving>;
+    fn as_trade_retrieving(self: Arc<Self>) -> Arc<dyn TradeRetrieving>;
+}
```

Comment:
Alternatively you can make handle_all_routes take `Arc<dyn TradeRetrieving>` and another `Arc<dyn PreSignatureRetrieving>` . It would just be a clone of the same arc of course but that way no intermediate trait is needed.

Reply:
I like the current method better, though I have to admit it looks a little bit strange. 

I like it better, as soon we might get other retrievers, like AppDataRetriever, etc, and cloning each time the Arc results in functions with a lot of different input parameters, all intended to talk to the same database.

But I don't have a strong opinion

Reply:
I discussed this with Valentin and the plan would be to use a concrete type for `database` (the instrumented database instead of `Arc<dyn DataRetrieving>`). I tried to do this but then I realized that the e2e tests use the original database, so to make this work I'd need some `NoopMetrics` for the orderbook metrics, which do not exist yet and would be too much code for this PR.
The plan now is to keep the PR as it is and in a follow-up PR introduce `NoopMetrics` and use a concrete type, which allows us to eventually remove this extra trait without extra cost.